index.js

```diff
@@ -3741,6 +3741,23 @@
   //. [type identifier][] is `'Object'` and the values of its enumerable own
   //. properties are all members of type `a`.
 
+  //# singleton :: String -> a -> StrMap a
+  //.
+  //. Embeds a key and a value into a string map
```

Comment:
This sounds more like `insert :: String -> a -> StrMap a -> StrMap a` to me. How about the following?

_Takes a string and a value of any type, and returns a string map with a single entry (mapping the key to the value)._